components-part1/src/partials/Home/Content.js

```diff
@@ -11,40 +13,99 @@ import {
   StyledResults,
   ContentWrapper,
 } from '../../styledComponents/Content';
-import {
-  fetchMoviesList,
-  filterByGenre,
-  sortByRating,
-  sortByReleaseDate,
-} from '../../actions/moviesActions';
+import httpService from '../../services/httpService';
 
 const Content = () => {
-  const dispatch = useDispatch();
-  const { movies, filteredMovies } = useSelector((state) => state);
+  const params = useParams();
+  const queryStr = useLocation().search;
+  const queryObj = queryString.parse(queryStr);
+  const { pathname } = useLocation();
+  const navigate = useNavigate();
+
   const [moviesList, setmoviesList] = useState([]);
-  const handleGenreClick = (genre) => {
-    dispatch(filterByGenre(movies, genre));
+  //this is the original list of movies
+  const [movies, setMovies] = useState(null);
+
+  const handleSortMovies = (movies, sortBy) => {
+    if (sortBy === 'vote_average') {
+      const sortedMovies = movies.sort(
+        (a, b) => b.vote_average - a.vote_average
+      );
+      return sortedMovies;
+    } else if (sortBy === 'release_date') {
+      const sortedMovies = movies.sort(
+        (a, b) => Date.parse(b.release_date) - Date.parse(a.release_date)
+      );
+      return sortedMovies;
+    }
   };
   useEffect(() => {
-    dispatch(fetchMoviesList('/movies'));
-  }, []);
+    const fetchData = async () => {
+      try {
+        const { data } = await httpService.get('/movies');
+        setMovies(data);
+        setmoviesList(data.data);
+      } catch (error) {
+        alert(error);
+      }
+    };
+    if (!queryStr && !params.searchQuery) {
+      fetchData();
+    }
+  }, [params, queryStr]);
+
   useEffect(() => {
-    if (movies.status === 'success') {
-      setmoviesList(movies.data.data);
+    /* if (movies && params.searchQuery) {
+      setmoviesList(handleSearchQuery(movies.data, params.searchQuery));
     }
-    if (filteredMovies.data) {
-      setmoviesList(filteredMovies.data);
+    if (movies && queryObj.genre) {
+      setmoviesList(handleSearchQuery(movies.data, queryObj.genre));
+    } */
+    if (queryObj.sortBy) {
+      const sortedMovies = handleSortMovies(moviesList, queryObj.sortBy);
+      setmoviesList(sortedMovies);
     }
-  }, [movies.status, filteredMovies.data]);
+  }, [queryObj.sortBy]);
```

Comment:
this Effect should also be applied when the movies have changed. Basically, the movies to show depend on 3 inputs: 'genre', 'sort', and 'search', should have them all in the dependency list.

Reply:
I kind of fixed that issue in the later PR, maybe you want to have a look. thanks.